pydatastructs/trees/binary_trees.py

```diff
@@ -415,6 +415,107 @@ def rank(self, x):
             walk = p
         return r
 
+    def _simple_path(self, key, root, path):
+        """
+        Utility funtion to find the simple path between root and node.
+        """
+        if root is None:
+            return False
+        path.append(root)
+        if self.tree[root].key == key:
+            return True
+
+        if self._simple_path(key, self.tree[root].left, path) or \
+            self._simple_path(key, self.tree[root].right, path):
+            path.append(root)
+            return True
+
+        path.pop()
+        return False
+
+    def simple_path(self, key):
+        path = []
+        self._simple_path(key, self.root_idx, path)
+        return path
```

Comment:
The logic here can be shifted to `_simple_path` as IMO, `simple_path` doesn't need to be public.

Reply:
Its true that we don't often use simple paths in BST. Are you suggesting to hide the simple_path function?

Reply:
I meant to keep all the logic for simple paths in `_simple_path`.

Reply:
Can you please tell in which lines is `simple_path` called? It is called more than once? Any research paper or Wikipedia or university lecture describing about simple paths? What are the possible use cases of simple paths other than finding LCA?

Reply:
I was also trying to expose the simple path function to users. That is the reason why I wrote one exposed and a hidden function. About simple path https://en.wikipedia.org/wiki/Path_(graph_theory) and CLRS book. Other use cases are to find the intermediate nodes between root and given node.

Reply:
Mainly used for LCA algorithm 2. So, should I remove it?

Reply:
Yes, put all the logic for finding simple paths under `_simple_path`.

Reply:
Finished merging.